lib/productions/operation.js

```diff
@@ -0,0 +1,46 @@
+import { Base } from "./base.js";
+import { return_type, argument_list, unescape } from "./helpers.js";
+
+export class Operation extends Base {
+  /**
+   * @param {import("../tokeniser.js").Tokeniser} tokeniser
+   */
+  static parse(tokeniser, { special, regular } = {}) {
+    const tokens = { special };
+    const ret = new Operation({ source: tokeniser.source, tokens });
+    if (special && special.value === "stringifier") {
+      tokens.termination = tokeniser.consume(";");
+      if (tokens.termination) {
+        ret.arguments = [];
+        return ret;
+      }
+    }
+    if (!special && !regular) {
+      tokens.special = tokeniser.consume("getter", "setter", "deleter");
+    }
+    ret.idlType = return_type(tokeniser) || tokeniser.error("Missing return type");
+    tokens.name = tokeniser.consume("identifier");
+    tokens.open = tokeniser.consume("(") || tokeniser.error("Invalid operation");
+    ret.arguments = argument_list(tokeniser);
+    tokens.close = tokeniser.consume(")") || tokeniser.error("Unterminated operation");
+    tokens.termination = tokeniser.consume(";") || tokeniser.error("Unterminated attribute");
```

Comment:
Unterminated operation as the error, no? Could be the same as above regarding the “;”.

Reply:
Oh wow, never noticed that mistake. Good catch.